test/Types.purs

```diff
@@ -94,10 +95,10 @@ instance showTree :: Show a => Show (Tree a) where
 instance eqTree :: Eq a => Eq (Tree a) where
   eq x y = genericEq x y
 
-instance decodeTree :: Decode a => Decode (Tree a) where
+instance decodeTree :: (Decode a, DecodeWithOptions a) => Decode (Tree a) where
```

Comment:
@jsparkes  Why do we need to add `DecodeWithOptions a` in shifting of compiler version. I am not able to understand this. Can you explain this